glean-core/android/src/main/java/mozilla/telemetry/glean/net/PingUploader.kt

```diff
@@ -69,9 +69,10 @@ interface PingUploader {
      * @param url the URL path to upload the data to
      * @param data the serialized text data to send
      * @param headers a [HeadersList] containing the headers to add.
+     * @param isGzipped whether or not the payload is gzipped
      *
      * @return return the status code of the upload response,
      *         or null in case upload could not be attempted at all.
      */
-    fun upload(url: String, data: String, headers: HeadersList): UploadResult
+    fun upload(url: String, data: ByteArray, headers: HeadersList, isGzipped: Boolean): UploadResult
```

Comment:
This is public API. Do we really want to break it _again_?
Why does the `upload` function need to know this?

Reply:
Do you mean `isGzipped`? We're already (unfortunately) breaking it because of the signature change. This should not be too scary, we own the concept-fetch bridge in a-c

Reply:
Ah right, we already break it.
Still, don't see the point of `isGzipped` in that function

Reply:
All right, dropped the gzip part :)